02_bubble/jimbotech/bubble.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sorted := make([]int, len(s))
+	copy(sorted, s)
+
+	var ss []int
+	for k := range sorted {
+		ss = sorted[:len(sorted)-k-1]
+		for i, val := range ss {
+			if val > sorted[i+1] {
```

Comment:
Using a subslice to range over in the inner loop is interesting, but it does make this a little less clear since the relationship between `val` and `sorted[i]` is obscured. The inner `if` written as:
```
if sorted[i] > sorted[i+1] {
    sorted[i], sorted[i+1] = sorted[i+1], sorted[i]
}
```
is very clear what it is doing under what condition.

This is a lab exercise so it is ok and fun to explore other ways to do things, but one of the goals of Go is to keep it simple, so do try not to be too clever unless there is some other upside.

Reply:
Done